src/P2P/Node.hs

```diff
@@ -546,31 +541,35 @@ findNextPeer conf node = do
             i <- randomR node (0, max 1 (length s) - 1)
```

Comment:
The use of `randomR` has no semantic effect on success or failure of the transaction. However, we expect a lot of contention on the PRNG variable because all threads make use of it.

Therefor the use of randomR should be isolated in a separate STM transaction, or, probably even better, be replaced by a concurrency mechanism that performance better under high contention.